ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala

```diff
@@ -441,21 +436,24 @@ trait H2OGridSearchParams extends H2OCommonSupervisedParams {
 
   def getSelectBestModelBy(): String = $(selectBestModelBy)
 
-  def getSelectBestModelDecreasing(): Boolean = $(selectBestModelDecreasing)
+  def getSelectBestModelOrdering(): String = $(selectBestModelOrdering)
 
   //
   // Setters
   //
 
   def setAlgo(value: H2OSupervisedAlgorithm[_, _, _ <: Model.Parameters]): this.type = {
+    val m = value.getClass.getDeclaredMethod("updateH2OParams")
```

Comment:
This was major bug in Grid search! We were passing algo to Grid on which it needs to perform the search, but the parameters for the model were still default before the method `updateH2OParams` which sets the paramters from spark wrapper to H2O parameter class wasn't called